app/views/events/participants.html.erb

```diff
@@ -1,40 +1,66 @@
+           });
+       } else {
+          $(':checkbox').each(function() {
+               this.checked = false;                        
+           });
+       } 
+    });
+  });
+</script>
 <div class="page-header">
   <h1><%=t '.title', :default => model_class.model_name.human.titleize %> (<%= @event.kind.humanize %>)</h1>
 </div>
 <% if can? :view_participants, Event %>
   <%= render :partial => 'show_event' %>
   
   <h1><%= t '.participants' %></h1>
-  <table class="table table-striped">
-    <tr>
-      <th><%= t 'activerecord.attributes.profile.name' %></th>
-      <th><%= t 'activerecord.attributes.profile.email' %></th>
-  	  <th><%= t 'agreement_letters.agreement_letter' %></th>
-    </tr>  
-  
-    <% @participants.each do |participant| %>
-    <tr>
-      
-      <td>
-      <%= participant.name %>
-      </td>
-      <td>
-      <%= participant.email %>
-      </td>
-      <td>
-        <% if participant.requires_agreement_letter_for_event?(@event) %>
-          <span class="text-muted"> <%= t '.unnecessary' %> </span>
-        <% elsif participant.agreement_letter_for_event?(@event) %>
-          <span class="text-success" title="<%= participant.agreement_letter_for_event(@event).updated_at %>"> <%= t '.available' %> </span>
-        <% else %>
-          <span class="text-danger"><%= t '.unavailable' %></span>
+  <%= form_tag(event_path(@event) + "/participants/agreement_letters") do %>
+    <table class="table table-striped">
+      <tr>
+        <th><%= t 'activerecord.attributes.profile.name' %></th>
+        <th><%= t 'activerecord.attributes.profile.email' %></th>
+    	  <th><%= t 'agreement_letters.agreement_letter' %></th>
+        <% if can? :print_agreement_letters, Event %><th><input type="checkbox" id="selectAll" value="selectAll" /> <%= t 'events.agreement_letters_download.select_all' %></th><% end %>
+      </tr>  
+    
+      <% @participants.each do |participant| %>
+      <tr>
+        
+        <td>
+        <%= participant.name %>
+        </td>
+        <td>
+        <%= participant.email %>
+        </td>
+        <td>
+          <% if participant.requires_agreement_letter_for_event?(@event) %>
+            <span class="text-muted"> <%= t '.unnecessary' %> </span>
+          <% elsif participant.agreement_letter_for_event?(@event) %>
+            <span class="text-success" title="<%= participant.agreement_letter_for_event(@event).updated_at %>"> <%= t '.available' %> </span>
+          <% else %>
+            <span class="text-danger"><%= t '.unavailable' %></span>
+          <% end %>
+        </td>
+        <% if can? :print_agreement_letters, Event %>
+          <td>
+            <%= check_box_tag('selected_participants[]', participant.id) %>
+          </td>
         <% end %>
-      </td>
-    </tr>
+      </tr>
+      <% end %>
+    </table>
+    <% if can? :print_agreement_letters, Event %>
+      <%= submit_tag t('events.agreement_letters_download.download_all_as'), name: "download_all" %>
```

Comment:
PO requested that there be only one button (I had two buttons implemented previously).